### PR TITLE
Issue 5738 - RFE - UI - Read/write replication monitor info to .dsrc

### PR DESCRIPTION
**Pull request by mreynolds389**
Description:

Allow UI to use the .dsrc replication monitor info, and also allow the UI to write new report configurations.  This prevents an admin from having to enter this information every time they want to run a report

relates: https://github.com/389ds/389-ds-base/issues/5738

### CODE REVIEW

**Comment by mreynolds389** [comment]
I'm going to add a CI test for the cli dsrc changes

File: src/cockpit/389-console/src/lib/monitor/replMonitor.jsx [hunk 1]
```diff
@@ -231,6 +195,55 @@ export class ReplMonitor extends React.Component {
...
+                        `Failed to get .dsrc information: ${errMsg.desc}`
+                    );
+                    this.setState({
+                        loadingDSRC: false,
+                    });
+                });
+    }
+
     componentDidMount() {
         if (this.state.initCreds) {
```
Comment (droideck) [inline src/cockpit/389-console/src/lib/monitor/replMonitor.jsx:248]: When I go to `Synchronization Report` while there's no replication configured, I get the next error:

```
CMD: ReplMonitor: add credentials during componentDidMount ==> dsconf -j ldapi://%2fvar%2frun%2fslapd-standalone1.socket config get nsslapd-port nsslapd-localhost nsslapd-rootdn

index.js:2 Uncaught TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
    at bV (index.js:2:1754263)
    at Function.<anonymous> (index.js:2:1763262)
    at G (cockpit.js:5:5607)
    at cockpit.js:5:5802
    at k (cockpit.js:5:4658)
```

**Review by droideck — CHANGES_REQUESTED** [review]
Another issue I faced is that `Saved Report Configuration` is not saved after I reload the page. (or if say it correctly, the list is not populated with .dsrc content after the reload)

File: src/cockpit/389-console/src/lib/monitor/replMonitor.jsx (hunk 1 above)
Comment (mreynolds389) [reply src/cockpit/389-console/src/lib/monitor/replMonitor.jsx:248]: Fixed

**Comment by mreynolds389** [comment]
> Another issue I faced is that `Saved Report Configuration` is not saved after I reload the page. (or if say it correctly, the list is not populated with .dsrc content after the reload)

Ugh of course a harmless last minute change broke this.  It's working now.

**Review by droideck — APPROVED** [review]
Looks great! Ack

P.S. I have a minor concern - `~/.dsrc` is not created automatically when it's absent (it just reports an error, so the user will need to go and create it manually).
But I don't have a strong opinion if we SHOULD create it. Probably, it's okay like that.

**Comment by mreynolds389** [comment]
> Looks great! Ack
> 
> P.S. I have a minor concern - `~/.dsrc` is not created automatically when it's absent (it just reports an error, so the user will need to go and create it manually).

No we should create it, makes it so much easier for an admin.  I'll work on that...

**Comment by mreynolds389** [comment]
This caused some regressions in the web UI testing, investigating:

dirsrvtests/tests/suites/webui/monitoring/monitoring_test.py:132:

**Pull request merged**